src/plugins/intel_cpu/src/nodes/reduce.cpp

```diff
@@ -2608,6 +2643,20 @@ inline void Reduce::create_working_memory() {
     dst_size = desc.get_size();
 }
 
+inline void Reduce::create_DH_working_memory() {
+    ReduceDH_opt = layout == ReduceLayoutType::reduce_nspc && support_split &&
+                   !ReduceC && ReduceD && ReduceH && !ReduceW && IC == 1 && ID > 1;
+    if (ReduceDH_opt) {
+        PD = ID;
+        PW = IW / blk_size * blk_size;
+        prc_data_size = src_data_size;
+        prc_size = PD * PW * src_data_size;
+        if (prc_size > vec_reduceDH_prc.size()) {
+            vec_reduceDH_prc.resize(prc_size);
```

Comment:
The major concern here: in dynamism case we will allocate new memory on each iteration which will result in huge perf penalties. 
I would suggest the following: 
1. Can we consider disabling such optimization for dynamic case as WA for 2022.2?
2. For 2022.3 we can modify the solution to use CPU memory manager for intermediate memory which automatically will reallocate memory only in case newly requested amount is bigger than available. 
What do you think?

Reply:
Fixed. Added !isDynamicNode() as one of the condition of the opt flag. Thanks Dmitry!